src/plugins/hetero/src/sync_infer_request.cpp

```diff
@@ -0,0 +1,175 @@
+        const auto& output_tensor = m_infer_requests[submodel_idx_out]._request->get_tensor(output_port);
+        const auto& input_port = m_infer_requests[submodel_idx_in]._network->inputs()[tensor_idx_in];
+        m_infer_requests[submodel_idx_in]._request->set_tensor(input_port, output_tensor);
+    }
+}
+
+ov::hetero::InferRequest::~InferRequest() = default;
+
+ov::SoPtr<ov::IAsyncInferRequest> ov::hetero::InferRequest::get_request(const ov::Output<const ov::Node>& port) const {
+    auto check_nodes = [](const ov::Node* node1, const ov::Node* node2) {
+        return node1 == node2 ||
+               (node1->get_friendly_name() == node2->get_friendly_name() &&
+                node1->get_type_info() == node2->get_type_info() &&
+                node1->outputs().size() == node2->outputs().size() && node1->inputs().size() == node2->inputs().size());
+    };
+
+    for (const auto& kvp : m_port_to_request_idx_map) {
+        if (kvp.first.get_index() == port.get_index() && kvp.first.get_names() == port.get_names() &&
+            check_nodes(kvp.first.get_node(), port.get_node())) {
+            return m_infer_requests[kvp.second]._request;
+        }
+    }
+    OPENVINO_THROW("Cannot find infer request for port ", port);
+}
+
+ov::Tensor ov::hetero::InferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
+    return get_request(port)->get_tensor(port);
+}
+
+void ov::hetero::InferRequest::set_tensor(const ov::Output<const ov::Node>& port, const ov::Tensor& tensor) {
+    get_request(port)->set_tensor(port, tensor);
+}
+
+std::vector<ov::Tensor> ov::hetero::InferRequest::get_tensors(const ov::Output<const ov::Node>& port) const {
+    return get_request(port)->get_tensors(port);
+}
+
+void ov::hetero::InferRequest::set_tensors(const ov::Output<const ov::Node>& port,
+                                        const std::vector<ov::Tensor>& tensors) {
+    return get_request(port)->set_tensors(port, tensors);
+}
+
+void ov::hetero::InferRequest::check_tensors() const {
+    // Ignore `check_tensor` of inputs and outputs of Hetero Compiled Model because
+    // `m_tensors` are not allocated
+    return;
+}
+
+std::vector<std::shared_ptr<ov::IVariableState>> ov::hetero::InferRequest::query_state() const {
+    std::vector<std::shared_ptr<ov::IVariableState>> variable_states = {};
+    for (auto&& desc : m_infer_requests) {
+        auto& r = desc._request;
+        assert(r);
+        for (auto&& state : r->query_state()) {
+            variable_states.emplace_back(state);
+        }
+    }
+    return variable_states;
+}
+
+std::shared_ptr<const ov::hetero::CompiledModel> ov::hetero::InferRequest::get_hetero_model() const {
+    auto& compiled_model = get_compiled_model();
+    auto hetero_model = std::dynamic_pointer_cast<const ov::hetero::CompiledModel>(compiled_model);
+    OPENVINO_ASSERT(hetero_model);
+    return hetero_model;
+}
+
+void ov::hetero::InferRequest::infer() {
+    start_pipeline();
+    wait_pipeline();
```

Comment:
You don't need wait pipeline. Here you use sync API for run.

Reply:
Removed. Also moved `cancel()` from sync infer request to async